### PR TITLE
Authentication refactor and cleanup

### PR DESCRIPTION
**Pull request by HeDo88TH**
Now storing users on database. We are using mysql/mariadb as backend. We could switch to sqlite for portability but it's not high in the priority list.
If the database does not exist It will be created as long as the admin user with "password" password.
I've included a default connection string in the configuration file, It should be changed to fit the deployment needs.

### CODE REVIEW

**Comment by pierotofy** [comment]
:+1: 

But, could we actually have SQLITE as the default?

The reason is that we want to make it easy for people to spin this up. A database setup (postgres, mysql) would be used for production/scalable uses and is an extra barrier/step for people to run this.

File: Registry.Adapters/FileSystem/PhysicalFileSystem.cs [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+        {
+            return Directory.Exists(path);
+        }
+
+        public IEnumerable<string> EnumerateFiles(string path, string searchPattern, SearchOption searchOption)
+        {
+            return Directory.EnumerateFiles(path, searchPattern, searchOption);
+        }
+
+        public IEnumerable<string> EnumerateFolders(string path, string searchPattern, SearchOption searchOption)
```
Comment (pierotofy) [inline Registry.Adapters/FileSystem/PhysicalFileSystem.cs:33]: Careful that S3 does not have a concept of a folder.

Comment (pierotofy) [reply Registry.Adapters/FileSystem/PhysicalFileSystem.cs:33]: https://github.com/connormanning/arbiter/blob/master/arbiter/driver.hpp has a good interface for reference.

Comment (HeDo88TH) [reply Registry.Adapters/FileSystem/PhysicalFileSystem.cs:33]: I'll integrate some changes in the interface to better adapt to S3 structure

Comment (HeDo88TH) [reply Registry.Adapters/FileSystem/PhysicalFileSystem.cs:33]: We can use Sqlite as default users backend

**Comment by HeDo88TH** [comment]
Added multi-provider support for identity with default to Sqlite + various fixes and improvements

**Comment by pierotofy** [comment]
Looking good!

**Pull request merged**